magpie/register.py

```diff
@@ -503,12 +503,18 @@ def _get_all_configs(path_or_dict, section):
     Loads all configuration files specified by the path (if a directory), a single configuration (if a file) or
     directly returns the specified dictionary section (if a configuration dictionary).
 
-    :returns: list of configurations loaded, list of single configuration if inputs wasn't a directory path.
+    :returns:
+        list of configurations loaded if input was a directory path
+        list of single configuration if input was a file path
+        list of single configuration if input was a JSON dict
+        empty list if none of the other cases where matched
     """
     if isinstance(path_or_dict, six.string_types) and os.path.isdir(path_or_dict):
         dir_path = os.path.abspath(path_or_dict)
         return [_load_config(os.path.join(dir_path, f), section) for f in os.listdir(dir_path) if f.endswith(".cfg")]
-    return [_load_config(path_or_dict, section)]
+    elif isinstance(path_or_dict, dict):
```

Comment:
I think you're missing another if case: `elif os.path.isfile(path_or_dict)`
If this function receives a path to a file, it will not be loaded.

Reply:
Indeed. Nice catch.